private/buf/cmd/buf/command/breaking/breaking.go

```diff
@@ -339,3 +351,103 @@ func validateFlags(flags *flags) error {
+		}
+		base = imageWithConfig.BreakingConfig()
+	}
+	return true
+}
+
+// Checks if the specified [bufconfig.BreakingConfig]s are equal. Returns true if both
+// [bufconfig.BreakingConfig] have the same configuration parameters.
+func equalBreakingConfig(breakingConfig1, breakingConfig2 bufconfig.BreakingConfig) bool {
+	if breakingConfig1.Disabled() != breakingConfig2.Disabled() {
+		return false
+	}
+	if breakingConfig1.FileVersion() != breakingConfig2.FileVersion() {
+		return false
+	}
+	if slices.Compare(breakingConfig1.UseIDsAndCategories(), breakingConfig2.UseIDsAndCategories()) != 0 {
+		return false
+	}
+	if slices.Compare(breakingConfig1.ExceptIDsAndCategories(), breakingConfig2.ExceptIDsAndCategories()) != 0 {
+		return false
+	}
+	if slices.Compare(breakingConfig1.IgnorePaths(), breakingConfig2.IgnorePaths()) != 0 {
+		return false
+	}
+	for ignoreID, paths1 := range breakingConfig1.IgnoreIDOrCategoryToPaths() {
+		paths2, ok := breakingConfig2.IgnoreIDOrCategoryToPaths()[ignoreID]
+		if !ok {
+			return false
+		}
+		if slices.Compare(paths1, paths2) != 0 {
+			return false
+		}
+	}
+	if breakingConfig1.DisableBuiltin() != breakingConfig2.DisableBuiltin() {
+		return false
+	}
+	return breakingConfig1.IgnoreUnstablePackages() == breakingConfig2.IgnoreUnstablePackages()
+}
+
+// A helper function for filtering out images from [imagesWithConfig] if there is no corresponding
+// image in [againstImages]. We determine this based on image file path.
+//
+// This assumes that len(imageWithConfigs) > len(againstImages).
+// We also expect that each image in [againstImages] is mapped only once to a single
+// imageWithConfig in [imagesWithConfig]. If an againstImage is found, then we don't check
+// it again. We also validate that each image in [againstImages] is mapped to an imageWithConfig
+// from [imageWithConfigs].
+func filterImagesNotInAgainstImages(
+	imageWithConfigs []bufctl.ImageWithConfig,
+	againstImages []bufimage.Image,
+) ([]bufctl.ImageWithConfig, error) {
+	foundAgainstImageIndices := make(map[int]struct{})
+	var filteredImageWithConfigs []bufctl.ImageWithConfig
+	for _, imageWithConfig := range imageWithConfigs {
+		for _, imageFile := range imageWithConfig.Files() {
+			var foundImage bufimage.Image
+			for i, againstImage := range againstImages {
+				if _, ok := foundAgainstImageIndices[i]; ok {
+					continue
+				}
+				if againstImage.GetFile(imageFile.Path()) != nil {
+					foundAgainstImageIndices[i] = struct{}{}
+					foundImage = againstImage
+					break
+				}
+			}
+			if foundImage != nil {
+				filteredImageWithConfigs = append(filteredImageWithConfigs, imageWithConfig)
+				break
+			}
+		}
+	}
+	// If we are unsuccessful in mapping all againstImages to a unique imageWithConfig, then
+	// we return the same
+	if len(foundAgainstImageIndices) != len(againstImages) || len(againstImages) != len(filteredImageWithConfigs) {
+		return nil, newInputAgainstImageCountError(len(imageWithConfigs), len(againstImages))
```

Comment:
Maybe this should be more informative that the image was deleted/unmatched (and is therefore a breaking change?).

Reply:
Hmm, so I waffled on whether I wanted to provide a different error message here or if simply providing the known error message was better. Since this isn't actually part of the check, I wasn't sure if returning an error message that appears to be a "breaking change" makes sense -- especially since it would only happen in a very specific subset of cases.